### ISSUE TITLE
C++ Builder incorrectly decodes header under the release configuration

### ISSUE THREAD

**Issue by 7even11** [IndySockets/Indy#399]
Attached with this bug report is a project which demonstrates it:
[IndyBug-CPP.zip](https://github.com/IndySockets/Indy/files/8040776/IndyBug-CPP.zip)

RAD studio used is **10.4.2** (Sydney)

Under the **Debug** configuration program builds and runs fine and properly decodes subject header (note - the same happens for other headers as well as filenames under the body):

```
Проба 123
```

Under the **Release** configuration, the output is garbled.

```
ðÒÏÂÁ 123
```
The only difference in release configuration are the usual to build standalone executable:
- C++ Linker / Link with dynamic RTL - unchecked
- Runtime packages / Link with runtime packages - unchecked.

The above is in KOI8-R (Russian) encoding but the problem doesn't only appear for that encoding, the same happens for windows-1255 (Hebrew) encoding as well and most likely other encodings.

Code:

```C++
TMemoryStream* ms = new TMemoryStream();
char buff[] = "Subject: =?KOI8-R?B?8NLPwsE=?= 123\r\n\r\ntest";
ms->Write(buff, sizeof(buff));
ms->Position = 0;

IdMsg->Clear();
// Suppress depreciated warning
#pragma warn -8111
Idmessagehelper::TIdMessageHelper_LoadFromStream(IdMsg, ms, false, false);
#pragma warn .8111

Memo1->Lines->Clear();
Memo1->Lines->Add(IdMsg->Subject);

delete ms;
```

So in other words, when exe is statically linked it doesn't work but it works from the debugger config.

Again, the problem occurs both on headers as well as body attachment filenames.

The problem has been present for a long time in Indy.
